Modules/Forum/classes/class.ilForum.php

```diff
@@ -622,7 +640,15 @@ public function moveThreads($thread_ids = array(), $src_ref_id = 0, $dest_top_fr
 					$objTmpThread->update();
 					
 					unset($objTmpThread);
-				}				
+				}
+
+				if (array() !== $errorMessages) {
+					\ilUtil::sendFailure(
+						implode("<br><br>", $errorMessages),
+						true
+					);
+					$this->ctrl->redirectByClass('ilObjForumGUI', 'showThreads');
```

Comment:
I wanted to avoid to return a value or pass a reference because this changes the behavior of this function significantly, but if the controller is not allowed in here this is the only chance.